marconi-sidechain/src/Marconi/Sidechain/Bootstrap.hs

```diff
@@ -41,50 +54,51 @@ import System.FilePath ((</>))
 -- | Run Sidechain indexers
 runSidechainIndexers :: ReaderT SidechainEnv IO ()
 runSidechainIndexers = do
-  env <- ask
-  cliArgs <- view sidechainCliArgs
+  CliArgs{..} <- view sidechainCliArgs
   trace <- view sidechainTrace
+  env <- view sidechainIndexersEnv
+
   let addressUtxoCallback :: State UtxoHandle -> IO ()
       addressUtxoCallback =
         atomically
-          . AddressUtxo.updateEnvState (env ^. sidechainIndexersEnv . sidechainAddressUtxoIndexer)
+          . AddressUtxo.updateEnvState (env ^. sidechainAddressUtxoIndexer)
   let epochStateCallback :: State EpochStateHandle -> IO ()
       epochStateCallback =
         atomically
           . EpochState.updateEnvState
-            (env ^. sidechainIndexersEnv . sidechainEpochStateIndexer . epochStateIndexerEnvIndexer)
+            (env ^. sidechainEpochStateIndexer . epochStateIndexerEnvIndexer)
   let mintBurnCallback :: State MintBurnHandle -> IO ()
-      dbPath = CLI.dbDir cliArgs
       mintBurnCallback =
         atomically
           . MintBurn.updateEnvState
-            (env ^. sidechainIndexersEnv . sidechainMintBurnIndexer . mintBurnIndexerEnvIndexer)
+            (env ^. sidechainMintBurnIndexer . mintBurnIndexerEnvIndexer)
       utxoIndexerConfig =
         UtxoIndexerConfig
-          { ucTargetAddresses = CLI.targetAddresses cliArgs
+          { ucTargetAddresses = targetAddresses
           , ucEnableUtxoTxOutRef = False --  we do not save scriptRef for sidechain
           }
       indexers =
         [
           ( utxoWorker addressUtxoCallback utxoIndexerConfig
-          , Just $ dbPath </> utxoDbName
+          , Just $ dbDir </> utxoDbName
           )
         ,
-          ( epochStateWorker (CLI.nodeConfigPath cliArgs) epochStateCallback
-          , Just $ dbPath </> epochStateDbName
+          ( epochStateWorker nodeConfigPath epochStateCallback
+          , Just $ dbDir </> epochStateDbName
           )
         ,
-          ( mintBurnWorker mintBurnCallback $ CLI.targetAssets cliArgs
-          , Just $ dbPath </> mintBurnDbName
+          ( mintBurnWorker mintBurnCallback targetAssets
+          , Just $ dbDir </> mintBurnDbName
           )
         ]
+
   lift $
     runIndexers
```

Comment:
I did consider this, but because `runIndexers` is in `marconi-chain-index` and `SidechainEnv` is in `marconi-sidechain` I thought it would break encapsulation.

Also, looking at the code for `runIndexers`, the uses of the parameters are spread throughout the function, so I don't think the use of `ReaderT` with a more specific env record would add much value. Also, I don't think the record would be reusable anywhere else.

However, if you can see advantages to it, I'm happy to give it a go.

Reply:
Yes, we can't use `SidechainEnv` as it would indeed create a dependency loop. And definitely, it isn't worth the effort if we need to split in the next calls.